internal/appstore/appstore.go

```diff
@@ -118,3 +136,76 @@ func createRequest(url, username, accesskey string, body *bytes.Buffer, contentT
 
 	return req, nil
 }
+
+// Locate looks for a file having the same signature.
+func (s *AppStore) Locate(filename string) (storage.ArtifactMeta, error) {
+	if filename == "" {
+		return storage.ArtifactMeta{}, nil
+	}
+
+	hash, err := calculateBundleHash(filename)
+	if err != nil {
+		return storage.ArtifactMeta{}, err
+	}
+
+	queryString := ""
+	for {
+		request, err := createLocateRequest(fmt.Sprintf("%s/v1/storage/list", s.URL), s.Username, s.AccessKey, queryString)
+		if err != nil {
+			return storage.ArtifactMeta{}, err
+		}
+
+		lr, err := s.executeLocateRequest(request)
+		if err != nil {
+			return storage.ArtifactMeta{}, err
+		}
+
+		for _, item := range lr.Items {
+			if item.ETag == hash {
+				return storage.ArtifactMeta{ID: item.ID}, nil
+			}
+		}
+
+		queryString = lr.Links.Next
+		if queryString == "" {
+			return storage.ArtifactMeta{}, nil
+		}
+	}
+}
+
+func calculateBundleHash(filename string) (string, error) {
+	fs, err := os.Open(filename)
+	if err != nil {
+		return "", err
+	}
```

Comment:
Not like this matters much in the workflow since the leak is small, but just in case:

```suggestion
	}
	defer fs.close()
```